src/Dotnet.Script/Program.cs

```diff
@@ -288,23 +273,45 @@ private static int Wain(string[] args)
                             code = SourceText.From(File.ReadAllText(absoluteSourcePath));
                         }
 
-                        // given the path to a script we create a %temp%\dotnet-scripts\{uniqueFolderName} path
-                        string pathToDll = Path.Combine(publishDirectory, Path.GetFileNameWithoutExtension(absoluteSourcePath) + ".dll");
-
                         // source hash is the checkSum of the code
-                        string sourceHash = Convert.ToBase64String(code.GetChecksum().ToArray());
+                        string sourceHash =
+                            BitConverter.ToString(code.GetChecksum().ToArray())
+                                        .Replace("-", string.Empty)
+                                        .ToLowerInvariant();
+
+                        string cacheFolder = Path.Combine(Path.GetTempPath(), "dotnet-scripts");
```

Comment:
let's use the folder name `dotnet-script`, as in the singular "product name"

Reply:
I think @tomlm meant it as plural but if we want to singular and product-based then how about `dotnet-script/cache`? It would enable other things might end up in the temporary directory to side under the same umbrella.